xwiki-platform-core/xwiki-platform-bridge/src/main/java/org/xwiki/bridge/event/DocumentMovedEvent.java

```diff
@@ -0,0 +1,62 @@
+/*
+  * See the NOTICE file distributed with this work for additional
+  * information regarding copyright ownership.
+  *
+  * This is free software; you can redistribute it and/or modify it
+  * under the terms of the GNU Lesser General Public License as
+  * published by the Free Software Foundation; either version 2.1 of
+  * the License, or (at your option) any later version.
+  *
+  * This software is distributed in the hope that it will be useful,
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  * Lesser General Public License for more details.
+  *
+  * You should have received a copy of the GNU Lesser General Public
+  * License along with this software; if not, write to the Free
+  * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+  * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+  */
+ package org.xwiki.bridge.event;
+ 
+ import org.xwiki.model.reference.DocumentReference;
+ import org.xwiki.observation.event.EndFoldEvent;
+ 
+ /**
+  * An event triggered after a document is moved.
+  * <p>
+  * The event also send the following parameters:
+  * </p>
+  * <ul>
+  * <li>data: the current {com.xpn.xwiki.XWikiContext} instance</li>
+  * </ul>
+  * 
+  * @version $Id$
+  * @since 10.0RC1
+  */ 
+ public class DocumentMovedEvent extends AbstractDocumentEvent implements EndFoldEvent
```

Comment:
As I said in a previous comment, you should not extend AbstractDocumentEvent for new events IMO. It's quite old and deprecated now (String based) plus it bring more complexity than simplification here (you don't have a specific document but two).

Reply:
I'm interested in this pull request. I have two questions:
- What class should DocumentMovedEvent extend instead of AbstractDocumentEvent? (I had a look at the existing document events and I could not find one extending a different class so far). 
- Since this request seems to be stalled, can I ask you for a hint on how to proceed? I guess I should merge the existing pull request into an xwiki-platform fork, modify it, and create a new pull request out of it, is that right?

Reply:
@tmortagne Why not deprecate `AbstractDocumentEvent` then and recommend to use `AbstractEntityEvent` instead then? Except that ideally we need to get away from oldcore which was the reason for introducing `AbstractDocumentEvent` in the bridge module initially ;)

Reply:
As I said documents events are based on AbstractDocumentEvent so it does not make much sense to deprecated it unless you plan to deprecated them too and introduce new events.

Reply:
I don't understand the reasoning: I don't see why deprecating `AbstractDocumentEvent` would mean deprecating classes extending it. Deprecating it is an indication that it shouldn't be used from now on.